application/di/init.go

```diff
@@ -123,7 +126,10 @@ func initInfrastructure() {
 }
 
 func initApplication() {
+	w := workspace.New(instrumentor, scanner, hoverService, scanNotifier) // don't use getters or it'll deadlock
```

Comment:
This makes me nervous - creating a workspace from inside di.Init(). But theoretically, it should work - previously, I believe, we created it when starting up a server, right?

Reply:
Previously it was done on the "initialize" handler (so after server creation), so I moved it a little earlier.

It doesn't have any side effect because the workspace doesn't depend on anything that is provided in the "initialize" message, and allows us to inject the workspace as a dependency to the service.

The workspace instance is never set to something else, so this is safe.